File: dashboard-front/app/scripts/service/deviceService.js
'use strict'
angular
  .module("dashboardFrontApp")
  .service('deviceService', ['$http', 'API', function ($http, API) {

    this.get = get;
    this.save = save;
    this.remove = remove;
    this.getAll = getAll;

    function get(id) {
      return $http.get(API.BASE_URL + "/device/" + id).then(function(response) {
        return response;
      }, function(error) {
        return error;
      });
    }

    function save(device) {
      var p;
      if (device.id) {
        // If update, use PUT method
        p = $http.put(API.BASE_URL + "/device", device);
      } else {
        // If new, use POST method
        p = $http.post(API.BASE_URL + "/device", device);
      }
      return p.then(function(response) {
        return response;
      }, function(error) {
        return error;
      });
    }

    function remove(id) {
      return $http.delete(API.BASE_URL + "/device/" + id)
        .then(function(response) {
          return response;
        }, function (error){
            return error;
        });
    }

    function getAll() {
      return $http.get(API.BASE_URL + "/device")
      .then(function(response) {
          return response;
      }, function (error){
        return error;
      });
    }

  }]);

File: dashboard-back/src/main/java/com/excilys/shooflers/dashboard/server/service/DeviceService.java
package com.excilys.shooflers.dashboard.server.service;

import com.excilys.shooflers.dashboard.server.model.Device;

import java.util.List;

public interface DeviceService {

  List<Device> getAll();

  /**
   * Create a device into DB.
   *
   * @param device the device
   */
  void save(Device device);

  /**
   * Update a device into DB.
   *
   * @param device the device to update
   * @return the new Feed or null if failed
   */
  void update(Device device);

  /**
   * Delete device from DB.
   *
   * @param uuid The device uuid
   */
  void delete(String uuid);

  /**
   * Get device from DB.
   *
   * @param uuid the device uuid
   * @return the new device or null if none
   */
  Device get(String uuid);

  /**
   * Register new device into DB.
   *
   * @param device the device to register
   */
  void register(Device device);

  /**
   * Removes all references to a specific feed.
   *
   * @param uuid the feed uuid
   */
  void removeFeed(String uuid);
}
